data/battle_scripts_1.s

```diff
@@ -8033,18 +8034,60 @@ BattleScript_SupremeOverlordActivates::
 
 BattleScript_CostarActivates::
 	pause B_WAIT_TIME_SHORT
-	call BattleScript_AbilityPopUp
+	call BattleScript_AbilityPopUpScripting
 	printstring STRINGID_PKMNCOPIEDSTATCHANGES
 	waitmessage B_WAIT_TIME_LONG
 	end3
 
 BattleScript_ZeroToHeroActivates::
 	pause B_WAIT_TIME_SHORT
-	call BattleScript_AbilityPopUp
+	call BattleScript_AbilityPopUpScripting
 	printstring STRINGID_ZEROTOHEROTRANSFORMATION
 	waitmessage B_WAIT_TIME_LONG
 	end3
 
+BattleScript_CommanderActivates::
+	saveattacker
```

Comment:
This doesn't actually save the attacker before Commander tries to do its thing (so if Dondozo or the Commander is switched in because of Eject Button, Eject Pack or multi target switching moves the attacker will become the Dondozo for the rest of the targets / hits), for a suggestion, maybe something like what I did here works better https://github.com/rh-hideout/pokeemerald-expansion/pull/5262/commits/f56aff7d7c9633b64de685aa13a6e4acfb1a842a (#5262)

Reply:
Yes, the saveattacker makes no sense so Instead I changed it to be saved before the attacker in code.I think it should be fine since this script should always end correctly?

Reply:
Though there was a different issue present that I fixed in this pr. See #4912

Reply:
Oh right, Eject Items only activate after hitting all targets. Nevermind, it should be fine now